modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -1258,23 +1259,26 @@ private boolean supportsTransactions(T componentModel) {
     return componentModel instanceof ConnectableComponentModel && ((ConnectableComponentModel) componentModel).isTransactional();
   }
 
+  private boolean requiresTransactionalActionConfiguration(T componentModel) {
+    return !componentModel.getModelProperty(NoTransactionalActionModelProperty.class).isPresent();
+  }
```

Comment:
I'm noticing we are checking the model property to determine if a CMP requires transactional configuration, which for me makes perfect sense.
For a moment I considered the possibility of checking for the existence of the parameter, but I guess we wouldn't want any config to be applied if, for some reason, a "transactionalAction" parameter is explicitly is added to a Reusable Operation.
I think what we are doing is better, but I also think we should add a test that describes what should happen in such case, right? Should we reject such operation definition? or should we just ignore it for transactional purposes? I'm more inclined towards the former, but I think right now we are doing the later.

Reply:
You mean to check if it's okey to have a parameter that is transactionalAction? I guess it would be confusing as a parameter name, but if that's what Daiana defines, it's on her. I can add this test to make sure it works!

Reply:
Yes, I think it will work right now. But the importance of this test is to document what would we like to happen in such case (and that we actually considered it). Because there are other alternatives, for example, forbidding it.

To be honest I don't know which approach is best. If we go for consistency with other SDKs I suppose we should allow it. But if we want to apply the "put the gorra now, you can always take it off later" philosophy, maybe we shouldn't.

We can discuss it in the spec if you want. And we can create another task if you want to defer it.